FV3/coarse_graining/coarse_graining.F90

```diff
@@ -693,6 +744,36 @@ subroutine masked_block_max_2d(fine, mask, coarse)
     enddo
    end subroutine masked_block_max_2d
 
+   subroutine block_edge_min_x(fine, coarse)
```

Comment:
Can you add a description for `block_edge_min_x` and `block_edge_min_y`? I think you're getting the minimum value of the fine resolution data mapped to the coarsened data, but I'm not totally sure.

Reply:
Sure thing, yeah, the signature of these probably looks a little peculiar, since they take in data on a mixed fine / coarse grid and output data on a coarse grid.  They are taking the minimum along a single axis of the fine-grid within coarse-grid cells. Since for the D-grid winds we do not use fine-grid data in the interior of coarse grid cells, we pre-downsample the fine-grid arrays along the axes we are not taking the minimum along before passing them to these subroutines to reduce the computational / memory cost.  I'll try and add a comment to that effect.